boards: add seeedstudio sensecap t1000e support (#22267)
RIOT-OS/RIOT: boards/seeedstudio-sensecap-t1000e/doc.md

| NRF52_AIN7           | P0.31   | NTC temperature ADC         |                               |
| T1000E_LORA_NSS_PIN  | P0.12   | LR1110 chip select          |                               |
| T1000E_LORA_RESET_PIN| P1.10   | LR1110 reset                |                               |
| T1000E_LORA_IRQ_PIN  | P1.01   | LR1110 IRQ (DIO1)           |                               |
| T1000E_LORA_BUSY_PIN | P0.07   | LR1110 busy (DIO2)          |                               |
| T1000E_GPS_EN_PIN    | P1.11   | GNSS power enable           |                               |
| T1000E_GPS_RESET_PIN | P1.15   | GNSS reset                  |                               |
| T1000E_3V3_EN_PIN    | P1.06   | Sensor 3V3 power enable     |                               |
| T1000E_3V3_ACC_EN_PIN| P1.07   | Accelerometer power enable  |                               |
| T1000E_ACC_INT_PIN   | P1.02   | QMA6100P interrupt          |                               |
| T1000E_CHRG_DETECT_PIN| P1.03  | Charging status             |                               |
| T1000E_EXT_PWR_DETECT_PIN| P0.05| External power detect      |                               |

## Flashing the Device

> The T1000-E has no onboard debugger. Use an external J-Link or OpenOCD-compatible

Reviewer: The Seeedstudio wiki states that it comes with the standard Adafruit nRF52 Bootloader as the Seeedstudio Xiao nRF52840 boards and the Adafruit Feather nRF52840 (and others) 🤔 

Accessing the SWD pins would require opening the device and overriding the bootloader would make the device unusable for the originally intended use cases.

Author: Yes you're right.

| T1000E_CHRG_DETECT_PIN| P1.03  | Charging status             |                               |
| T1000E_EXT_PWR_DETECT_PIN| P0.05| External power detect      |                               |

## Flashing the Device

The T1000-E ships with the
[Adafruit nRF52 Bootloader](https://github.com/adafruit/Adafruit_nRF52_Bootloader),
you can find flashing instructions
[Adafruit nRF52 Bootloader](@ref boards_common_adafruit-nrf52-bootloader) section.

DFU mode must be entered manually before flashing. Follow the
[SeeedStudio instructions](https://files.seeedstudio.com/wiki/SenseCAP/Meshtastic/dfu-mode2.gif)
to put the device in DFU mode, then run:

```shell
BOARD=seeedstudio-sensecap-t1000e make -C examples/basic/hello-world flash
```

@note Accessing the SWD pads requires opening the device enclosure. Overwriting
the bootloader via SWD would make UF2 flashing unavailable and is not recommended
for normal use.

## Accessing STDIO

By default, `stdio` uses `UART_DEV(1)` (P0.16/P0.17).
`UART_DEV(0)` is reserved for the GNSS module. Connect a USB-to-UART adapter
to P0.16 (TX) and P0.17 (RX) to access the terminal.

> To use USB CDC ACM (virtual serial port over the existing USB cable) instead:

Reviewer: ```suggestion
To use USB CDC ACM (virtual serial port over the existing USB cable) instead,
you can add `USEMODULE+=stdio_cdc_acm` to your application's Makefile or to
the compiler call:
```

@defgroup   boards_seeedstudio-sensecap-t1000e SeeedStudio SenseCAP T1000-E
@ingroup    boards
@brief      Support for the SeeedStudio SenseCAP T1000-E
@author     Baptiste Le Duc <baptiste.leduc@etik.com>

Pin configuration is based on the
[Meshtastic firmware variant](https://github.com/meshtastic/firmware/blob/develop/variants/nrf52840/tracker-t1000-e/variant.h)
for this board.

## Overview

The [SenseCAP T1000-E](https://wiki.seeedstudio.com/sensecap_t1000_e/) is a
compact LoRaWAN tracker based on the nRF52840 MCU with the following
on-board components:

- Nordic nRF52840 SoC @64MHz, 1MB Flash, 256KB RAM
- Semtech LR1110 LoRa transceiver
- Mediatek AG3335 GNSS module
- QMA6100P 3-axis accelerometer
- NTC temperature sensor
- Light sensor
- Buzzer
- 1 user button
- 1 LED
- LiPo battery support with charging detection

## Hardware

| MCU         | nRF52840                               | Supported |
|:----------- |:-------------------------------------- |:---------:|

> | Family      | ARM Cortex-M4F                         |           |

Reviewer: ```suggestion
| Family      | ARM Cortex-M4 with FPU                 |           |
```

Perhaps, to preempt the effort in #22298.

Background: `M4F` is not actually a term used by ARM. While there are M4 variants with and without FPU, they are not distinguished by an `F` in the official documentation.

| NRF52_AIN7           | P0.31   | NTC temperature ADC         |                               |
| T1000E_LORA_NSS_PIN  | P0.12   | LR1110 chip select          |                               |
| T1000E_LORA_RESET_PIN| P1.10   | LR1110 reset                |                               |
| T1000E_LORA_IRQ_PIN  | P1.01   | LR1110 IRQ (DIO1)           |                               |
| T1000E_LORA_BUSY_PIN | P0.07   | LR1110 busy (DIO2)          |                               |
| T1000E_GPS_EN_PIN    | P1.11   | GNSS power enable           |                               |
| T1000E_GPS_RESET_PIN | P1.15   | GNSS reset                  |                               |
| T1000E_3V3_EN_PIN    | P1.06   | Sensor 3V3 power enable     |                               |
| T1000E_3V3_ACC_EN_PIN| P1.07   | Accelerometer power enable  |                               |
| T1000E_ACC_INT_PIN   | P1.02   | QMA6100P interrupt          |                               |
| T1000E_CHRG_DETECT_PIN| P1.03  | Charging status             |                               |
| T1000E_EXT_PWR_DETECT_PIN| P0.05| External power detect      |                               |

## Flashing the Device

> The T1000-E ships with the
> [Adafruit nRF52 Bootloader](https://github.com/adafruit/Adafruit_nRF52_Bootloader),
> the same bootloader used on the Adafruit Feather nRF52840 and SeeedStudio XIAO
> nRF52840. Flashing is done via UF2 over USB.
> 
> DFU mode must be entered manually before flashing. Follow the
> [SeeedStudio instructions](https://files.seeedstudio.com/wiki/SenseCAP/Meshtastic/dfu-mode2.gif)
> to put the device in DFU mode, then run:
> 
> ```
> BOARD=seeedstudio-sensecap-t1000e make -C examples/basic/hello-world flash
> ```

Reviewer: `````suggestion
The T1000-E ships with the
[Adafruit nRF52 Bootloader](https://github.com/adafruit/Adafruit_nRF52_Bootloader),
you can find flashing instructions
[Adafruit nRF52 Bootloader](@ref boards_common_adafruit-nrf52-bootloader) section.

DFU mode must be entered manually before flashing. Follow the
[SeeedStudio instructions](https://files.seeedstudio.com/wiki/SenseCAP/Meshtastic/dfu-mode2.gif)
to put the device in DFU mode, then run:

```shell
BOARD=seeedstudio-sensecap-t1000e make -C examples/basic/hello-world flash
```
`````

Fun fact: we have a whole page dedicated to this bootloader :D

@note Accessing the SWD pads requires opening the device enclosure. Overwriting
the bootloader via SWD would make UF2 flashing unavailable and is not recommended
for normal use.

## Accessing STDIO

By default, `stdio` uses `UART_DEV(1)` (P0.16/P0.17).
`UART_DEV(0)` is reserved for the GNSS module. Connect a USB-to-UART adapter
to P0.16 (TX) and P0.17 (RX) to access the terminal.

To use USB CDC ACM (virtual serial port over the existing USB cable) instead:

> ```

Reviewer: `````suggestion
```shell
`````

USEMODULE=stdio_cdc_acm BOARD=seeedstudio-sensecap-t1000e make -C examples/basic/hello-world flash term
```

> 

Reviewer: ```suggestion
```

To make the static test happy.
